### ISSUE TITLE
Possible way to easilly suport stack multi-project setup with a simple change

### ISSUE THREAD

**Issue by rvion** [atom-haskell/ide-haskell#156]
ide-haskell / ghc-mod do not currently support multi package projects.
I think there is a way to change that easilly:
could you have one ghc-mod session per each root project folder ?
http://blog.atom.io/2015/04/15/multi-folder-projects.html

(following sentence is not a rant, just a status update, and I really thanks you for the ide-haskell and ghc-mod project) As of now, I can barely use ide-haskell / ghc-mod on my project, and find myself working against ghc slowness / other tools limitations half of the time (~10 packages / ~200 modules) everything is slow, tools crash when run simultaneously, etc. arf

**Comment by lierdakil** [comment]
It should already do that:

![image](https://cloud.githubusercontent.com/assets/7275622/15008304/0a96ca4e-11e8-11e6-970e-5473d0263af1.png)

There's a caveat however. If your root project folder doesn't contain cabal files, ghc-mod will search directory tree upwards and bind where it finds one (if any), which is probably not exactly what you expect.

**Comment by rvion** [comment]
Indeed, it seems to work Well ! I'm not able to reproduce the error I experienced last time I tried this.

> There's a caveat however. If your root project folder doesn't contain cabal files, ghc-mod will search directory tree upwards and bind where it finds one (if any), which is probably not exactly what you expect.

I'm think I'm hitting this case: in my top level folder, I have ~15 sub folder, ~10 being haskell project, ~5 being other things (js, etc). but I also have a .cabal directly in the top level folder. It means that when I add all ~15 folders, the 5 without .cabal will default to the same top level one. I guess I can just remove it for now
